data/api/server-server/user_devices.yaml

```diff
@@ -38,81 +27,89 @@ paths:
       parameters:
         - in: path
           name: userId
-          type: string
           required: true
           description: |-
             The user ID to retrieve devices for. Must be a user local to the
             receiving homeserver.
-          x-example: "@alice:example.org"
+          example: "@alice:example.org"
+          schema:
+            type: string
       responses:
-        200:
+        "200":
           description: The user's devices.
-          schema:
-            type: object
-            properties:
-              user_id:
-                type: string
-                description: The user ID devices were requested for.
-                example: "@alice:example.org"
-              stream_id:
-                type: integer
-                description: |-
-                  A unique ID for a given user_id which describes the version of
-                  the returned device list.  This is matched with the `stream_id`
-                  field in `m.device_list_update` EDUs in order to incrementally
-                  update the returned device_list.
-                example: 5
-              devices:
-                type: array
-                description: The user's devices. May be empty.
-                items:
-                  type: object
-                  title: User Device
-                  properties:
-                    device_id:
-                      type: string
-                      description: The device ID.
-                      example: "JLAFKJWSCS"
-                    keys:
-                      type: object
-                      description: Identity keys for the device.
-                      $ref: "../client-server/definitions/device_keys.yaml"
-                    device_display_name:
-                      type: string
-                      description: Optional display name for the device.
-                      example: "Alice's Mobile Phone"
-                  required: ['device_id', 'keys']
-              master_key:
+          content:
+            application/json:
+              schema:
                 type: object
-                description: |-
-                  The user\'s master cross-signing key.
-                allOf:
-                  - $ref: ../client-server/definitions/cross_signing_key.yaml
-                  # FIXME: why isn't the doc generator picking up this example?
```

Comment:
I haven't checked whether the doc generator is still not picking up the example but it might well not because `allOf` should be wrapped in `schema` here.

Reply:
The comment is gone on `main` and it seems the example is picked up in the spec.

I don't think `allOf` needs to be wrapped in a `schema` since this already is in the `schema` of the response. However it seems wrong to me that the `type` and `description` are outside of the `allOf`.

Reply:
Unlike `$ref`, `allOf` doesn't forbid sibling keys; but I agree that it would be better to put the entire schema inside `allOf` for integrity. Also, `type: object` is not needed then as it's already inherited from (in this case) `cross_signing_key.yaml`.

Reply:
Looking at examples in other places, having `type` and `description` outside of `allOf` is actually common and overrides whatever `allOf` comes with. I would not keep `type` because one should already come from `$ref` but that's not a big deal (and I see quite a bit of it across our APIs... so worth a separate cleanup PR, before or after the migration).